src/ast.scm

```diff
@@ -358,6 +358,20 @@
   (and (if one (length= e 3) (length> e 2))
        (eq? (car e) 'meta) (eq? (cadr e) 'nospecialize)))
 
+(define (if-generated? e)
+  (and (length= e 4) (eq? (car e) 'if) (equal? (cadr e) '(generated))))
+
+(define (generator-meta? e)
+  (and (length= e 3) (eq? (car e) 'meta) (eq? (cadr e) 'generator)))
```

Comment:
It seems a little confusing to have `Expr(:meta, :generator)`, which is a completely different and unrelated thing from `Expr(:generator, ...)`

Reply:
Fair point. I could use `generated` instead.